### PR TITLE
Fix/linting issues

### PR DESCRIPTION
**Pull request by sericaia**
React hooks useEffect dependencies is not being properly handled.

This commits adds linting for react hooks and fixes the current useEffect's problems.

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -21,6 +28,9 @@
   "rules": {
     "@typescript-eslint/explicit-function-return-type": "off",
     "@typescript-eslint/no-var-requires": "error",
+    "react/no-string-refs": "warn",
```
Comment (goldins) [inline .eslintrc.json:31]: 👍  good addition too

File: src/utility/icons-wrapper/IconsWrapper.tsx [hunk 2]
```diff
@@ -25,39 +25,45 @@ const successTooltip: Props = {
   placement: 'right'
 };
 
+interface IconWrapperProps {
+  child: React.ReactElement;
+}
+
+const IconWrapper = ({ child }: IconWrapperProps) => {
```
Comment (goldins) [inline src/utility/icons-wrapper/IconsWrapper.tsx:32]: IMO wrapping `IconWrapper` in `memo` would be cleaner than using two `useCallback`s. Performance gains might be negligible in either case though. (out of scope; up to you if you want to make changes)

File: src/components/accordion/Accordion.tsx [hunk 3]
```diff
@@ -39,7 +39,7 @@ const Accordion = ({
     } else {
       setOpenedSections(new Set<number>(opened));
     }
-  }, [opened]);
+  }, [opened, allSectionsOpened, children, oneSectionAlwaysOpen]);
```
Comment (sanatshah) [inline src/components/accordion/Accordion.tsx:42]: Thanks for this!

File: src/components/dropdown/DropdownMenu.tsx [hunk 4]
```diff
@@ -75,22 +84,19 @@ export const DropdownMenu: React.FC<Props> = props => {
       console.error('Error while computing DropdownMenu position', e);
       setPlacement(DEFAULT_PLACEMENT);
     }
-  });
+  }, [containerRef, controllerRef, dropdownPlacement, isOpen]);
```
Comment (sericaia) [inline src/components/dropdown/DropdownMenu.tsx:87]: I'm not 100% sure if we shouldn't add `menuRef` here too (linter does not complain). Any thoughts @adelego ?

File: src/components/dropdown/DropdownMenu.tsx [hunk 5]
```diff
@@ -38,34 +39,42 @@ const MenuWrapper = styled.div<MenuWrapperProps>`
 const DEFAULT_PLACEMENT = 'bottom';
 const POSSIBLE_PLACEMENTS: Placement[] = ['top', 'bottom'];
 
-export const DropdownMenu: React.FC<Props> = props => {
+export const DropdownMenu = ({
```
Comment (adelego) [inline src/components/dropdown/DropdownMenu.tsx:42]: I think it's better to explicitly type components. This way if we make a mistake with the return type, typescript will point it out.
Also, you won't need to explicit the `children` props

File: src/components/dropdown/DropdownMenu.tsx (hunk 4 above)
Comment (adelego) [reply src/components/dropdown/DropdownMenu.tsx:87]: Yes it would be better I think

File: src/utility/icons-wrapper/IconsWrapper.tsx (hunk 2 above)
Comment (ni0ki) [reply src/utility/icons-wrapper/IconsWrapper.tsx:32]: On this, I'm still learning when to use one, the other, or not use them at all, thanks for the advice :+1:
Might have wrote this a bit too fast.

File: src/components/dropdown/DropdownMenu.tsx (hunk 5 above)
Comment (sericaia) [reply src/components/dropdown/DropdownMenu.tsx:42]: I'm typing them we way we do it in another projects. I believe I can use `React.PropsWithChildren` instead.

**Pull request merged**